Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py

```diff
@@ -4131,6 +4131,123 @@ def get_cve_command(args: dict) -> list[CommandResults]:
+    remove_nulls_from_dictionary(query_params)
+    # Temporary solution:
+    return http_request('GET', f'/ods/queries/scans/v1?{"&".join(f"{k}={v}" for k,v in query_params.items())}')
+    # return http_request('GET', '/ods/queries/scans/v1', params=query_params)
+    
+
+def ODS_get_scans_by_id_request(ids: list[str]) -> dict:
+    if not ids:
+        raise DemistoException('No IDs to check.') # TODO  check if better response needed
+    url_params = '&'.join(f'ids={query_id}' for query_id in ids)
+    return http_request('GET', f'/ods/entities/scans/v1?{url_params}')
+
+
+def ODS_get_scan_resources_to_human_readable(resources: list[dict]) -> str:
+    # TODO ask if to use default headers for UI
+    
+    human_readable = tableToMarkdown(
+        'CrowdStrike Falcon ODS Scans',
+        [
+            {
+                'ID': scan_info.get('id'),
+                'Status': scan_info.get('status'),
+                'Severity': '--',  # TODO ask if needed
+                'Hosts with detections': sum(1 for host in scan_info.get('metadata', [{}]) if host.get('filecount', {}).get('malicious')),  # TODO omly in UI response
+                'Hosts targeted': len(scan_info.get('metadata', [])),
+                'Incomplete hosts': sum(1 for host in scan_info.get('metadata', [{}]) if host.get('status') != 'completed'),
+                'Description': scan_info.get('description'),
+                'File paths': scan_info.get('scan_inclusions'),
+                'Maximum CPU utilization': CPU_UTIL_INT_TO_HR.get(scan_info.get('cpu_priority')),
+                'Hosts/Host groups': scan_info.get('hosts') or scan_info.get('host_groups'),
+                'Start time': scan_info.get('scan_started_on'),
+                'End time': scan_info.get('scan_completed_on'),
+                'Run by': scan_info.get('created_by')
+            }
+            for scan_info in resources
+        ],
+        headers=['ID', 'Status', 'Severity', 'Hosts with detections',
+                 'Hosts targeted', 'Incomplete hosts', 'Description',
+                 'File paths', 'Maximum CPU utilization',
+                 'Hosts/Host groups', 'End time', 'Start time', 'Run by']
+    )
+    
+    return human_readable
+
+
+def build_cs_falcon_filter(custom_filter: str = None, **filter_args) -> str:
+    """Creates an FQL syntax filter from a dictionary and a custom built filter
+
+    :custom_filter: custom filter from user (will take priority if conflicts with dictionary), defaults to None
+    :filter_args: args to translate to FQL format.
+    
+    :return: FQL syntax filter.
+    """
+    remove_nulls_from_dictionary(filter_args)
+    result = (
+        (custom_filter.replace('+', '%2B') if custom_filter else '')
+        + ('%2B' if custom_filter and filter_args else '')
+        + ('%2B'.join(f'{key}:{argToList(value)}' for key, value in filter_args.items()))
+    )
+    return result
+
+
+def get_ODS_ids(args: dict) -> list[str]:
+
```

Comment:
lets add a debug statement here that we're fetching based on query api

Reply:
changed, pending confirmation